app/grandchallenge/reader_studies/forms.py

```diff
@@ -299,6 +312,21 @@ def __init__(self, *args, **kwargs):
             )
         )
 
+    def save(self, *args, **kwargs):
+        instance = super().save(*args, **kwargs)
+        if "widget" in self.changed_data:
+            new_widget = self.cleaned_data["widget"]
+            try:
+                old_widget = instance.widget
+            except ObjectDoesNotExist:
+                old_widget = None
+            if old_widget.pk:
+                old_widget.delete()
+            if new_widget:
+                new_widget.save()
+                instance.widget = new_widget
```

Comment:
Why is `instance.widget = new_widget` necessary? I would have though that the changes would automatically get picked up.

Reply:
Ah, you're right, this is not necessary!